domain/removal/service/application_test.go

```diff
@@ -373,6 +304,44 @@ func (s *applicationSuite) TestExecuteJobForApplicationDyingJujuSecretsDeleteApp
 	c.Assert(err, tc.ErrorIsNil)
 }
 
+func (s *applicationSuite) TestExecuteJobForApplicationDyingExternalSecretsDeleteApplication(c *tc.C) {
+	defer s.setupMocks(c).Finish()
+
+	j := newApplicationJob(c)
+
+	secretExternalRefs := []string{"wun", "too", "free"}
+
+	exp := s.modelState.EXPECT()
+	exp.GetApplicationLife(gomock.Any(), j.EntityUUID).Return(life.Dying, nil)
+	exp.GetApplicationOwnedSecretRevisionRefs(gomock.Any(), j.EntityUUID).Return(secretExternalRefs, nil)
+	exp.DeleteApplicationOwnedSecrets(gomock.Any(), j.EntityUUID).Return(nil)
+	exp.GetCharmForApplication(gomock.Any(), j.EntityUUID).Return(tc.Must(c, coreapplication.NewUUID).String(), nil)
+	exp.DeleteCharmIfUnused(gomock.Any(), gomock.Any()).Return(errors.Errorf("the charm is still in use"))
```

Comment:
I am worried about this guy. We don't share charms between applications. Should this not just be getting deleted with the application?